pkg/volume/glusterfs/glusterfs.go

```diff
@@ -604,6 +605,15 @@ func (d *glusterfsVolumeDeleter) getGid() (int, bool, error) {
 }
 
 func (d *glusterfsVolumeDeleter) Delete() error {
+
+	pvSpec := d.spec.Spec
+	retainPolicy := string(api.PersistentVolumeReclaimRetain)
+	reclaimPolicy := string(pvSpec.PersistentVolumeReclaimPolicy)
+	if reclaimPolicy == retainPolicy {
+		glog.V(1).Infof("glusterfs: [%v] PV retained successfully ", pvSpec.ClaimRef.Name)
```

Comment:
you don't need to check for this, if it's Retain, Delete() will never get called by pv controller